src/mbgl/shader/circle_uniforms.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <mbgl/shader/uniforms.hpp>
+
+namespace mbgl {
+
+MBGL_DEFINE_UNIFORM_SCALAR(float, u_radius);
+MBGL_DEFINE_UNIFORM_SCALAR(bool, u_scale_with_map);
+MBGL_DEFINE_UNIFORM_SCALAR(float, u_devicepixelratio);
+
+struct CircleUniforms : gl::Uniforms<
+    u_matrix,
+    u_opacity,
+    u_color,
+    u_radius,
+    u_blur,
+    u_scale_with_map,
+    u_extrude_scale,
+    u_devicepixelratio> {};
+
+} // namespace mbgl
```

Comment:
There's not a one-to-one relationship -- e.g. `FillShader`, `FillPatternShader`, `FillOutlineShader`, `FillOutlinePatternShader`. I think there's some consolidation to be done -- perhaps replacing explicit shader subclasses with a single `Shader` template -- but I'd like to defer it to a future PR.